File: test-projects/csharp-sample/Greeter.cs
namespace CSharpSample
{
    public class Greeter
    {
        private string greeting;

        public Greeter(string greeting)
        {
            this.greeting = greeting;
        }

        public string GreetPerson(string name)
        {
            return $"{greeting}, {name}!";
        }

        public decimal CalculateDiscount(decimal amount, int percent)
        {
            return amount * percent / 100;
        }

        public static List<int> GenerateRange(int start, int end)
        {
            var list = new List<int>();
            for (int i = start; i <= end; i++)
            {
                list.Add(i);
            }
            return list;
        }
    }
}

File: test-projects/typescript-sample/src/greeter.ts
export function greet(name: string): string {
  return `Hello, ${name}!`;
}

export class Greeter {
  constructor(private greeting: string) {}

  public greetPerson(name: string): string {
    return `${this.greeting}, ${name}!`;
  }
}

export function calculateDiscount(amount: number, percent: number): number {
  return amount * (percent / 100);
}

export function sumNumbers(numbers: number[]): number {
  return numbers.reduce((acc, curr) => acc + curr, 0);
}
